File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.Arrays;
class Solution {
    public boolean solution(String[] phone_book) {
        Arrays.sort(phone_book);
        for (int i=0; i<phone_book.length-1; i++) {
            int len = phone_book[i].length();
            if (phone_book[i+1].length()>=len && phone_book[i+1].substring(0,len).equals(phone_book[i]))
                return false;
        }
        return true;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.js
function solution(phone_book) {
    phone_book.sort();
    
    for (let i=0; i<phone_book.length-1; i++) {
        const phone_len = phone_book[i].length;
        if (phone_book[i+1].length>phone_len && phone_book[i] === phone_book[i+1].slice(0, phone_len)) {
            return false;
        }
    }
    return true;
}